File: 백준/Gold/1922. 네트워크 연결/네트워크 연결.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;
#define IAMFAST ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);

int n,m;
int a,b,c;
struct KS
{
    int s,e,v;
};
bool comp(KS x, KS y)
{
    return x.v < y.v;
}
vector<KS> v;
int parent[1001];

void INPUT()
{
    IAMFAST
    cin >> n >> m;
    for (int i = 0; i < m; i++)
    {
        cin >> a >> b >> c;
        v.push_back({a, b, c});
    }
}

int Find(int x)
{
    if(x == parent[x]) return x;
    else return parent[x] = Find(parent[x]);
}

bool Union(int x, int y)
{
    x = Find(x);
    y = Find(y);

    if(x == y) return true;

    parent[x] = y;
    return false;
}

void SOLVE()
{
    sort(v.begin(),v.end(),comp);

    for(int i = 1; i <= n; i++) parent[i] = i;

    int ans = 0;
    for(int i = 0; i < m; i++)
        if(!Union(v[i].s, v[i].e))
            ans += v[i].v;
    cout << ans;
}

int main()
{
    INPUT();
    SOLVE();
}

File: 백준/Gold/1922. 네트워크 연결/네트워크 연결.py
n = int(input())
m = int(input())
graph = []
ans = 0

for _ in range(m):
    u, v, w = map(int, input().split())
    graph.append((u, v, w))

parent = [i for i in range(n + 1)]

def find(x):
    if parent[x] == x:
        return x
    else:
        parent[x] = find(parent[x])
        return parent[x]

def union(x, y):
    x = find(x)
    y = find(y)

    if x == y:
        return False

    if x > y:
        parent[x] = y
    else:
        parent[y] = x

    return True

graph.sort(key=lambda x: x[2])

for g in graph:
    if union(g[0], g[1]):
        ans += g[2]

print(ans)